platform/mbed_application.c

```diff
@@ -170,7 +177,11 @@ __asm static void start_new_application(void *sp, void *pc)
 void start_new_application(void *sp, void *pc)
 {
     __asm volatile(
-        "movw   r2, #0      \n" // Fail to compile "mov r2, #0" with ARMC6. Replace with MOVW.
+#if defined(__CORTEX_M0PLUS)
+        "mov   r2, #0      \n" // No MOVW instruction on Cortex-M0+
```

Comment:
interesting note that ARMC6 fails, does this compile for m0+ ?

Reply:
I don't have armc6 to test. This was a comment from @ccli8. I also thought it was interesting, because this is part of the code path for IAR and GCC, not Arm compiler, so I don't know the significance of the comment. I just left it as is though.

Reply:
I see, @ccli8 should that comment be removed or ?

Reply:
@sarahmarshy ARMC6 (not ARM) toolchain doesn't define `__CC_ARM`, but define `__GNUC__`, so ARMC6 will compile this code path.

@0xc0170 I re-compile this code path with ARMC6 (6.10) and GCC_ARM (2018/Q2), `mov r2, #0` compiles failed only on ARMC6/M23 combination. So if the comment is to be left, it can change to more accurate like `Fail to compile "mov r2, #0" with ARMC6/M23. Replace with MOVW.`

```C
#elif defined (__GNUC__) || defined (__ICCARM__)

void start_new_application(void *sp, void *pc)
{
    __asm volatile(
        "movw   r2, #0      \n" // Fail to compile "mov r2, #0" with ARMC6. Replace with MOVW.
```

Reply:
Thanks @ccli8 . The latest update took care of it